### PR TITLE
Add test case for etcd watch

### PR DESCRIPTION
**Pull request by mrwwheat7**

### CODE REVIEW

**Review by pzheng1025 — DISMISSED** [review]

**Review by pwang7 — APPROVED** [review]

File: etcd-client/src/lib.rs [hunk 1]
```diff
@@ -220,10 +221,50 @@ mod tests {
...
+            .kv()
+            .put(EtcdPutRequest::new("41_foo1", "baz3"))
+            .await?;
+        let mut watch_response = response
+            .next()
+            .await
+            .unwrap_or_else(|| panic!("Fail to receive reponse from client"))
+            .unwrap_or_else(|e| panic!("failed to get watch response, the error is: {}", e));
+
+        assert_eq!(watch_response.watch_id(), 0);
```
Comment (pwang7) [inline etcd-client/src/lib.rs]: assert_eq!(watch_response.watch_id(), 0, "add comments to this assert");

File: etcd-client/src/lib.rs [hunk 2]
```diff
@@ -220,10 +221,50 @@ mod tests {
...
+        assert_eq!(watch_response.take_events().len(), 1);
```
Comment (pwang7) [inline etcd-client/src/lib.rs]: same here

**Review by pwang7 — CHANGES_REQUESTED** [review]

File: etcd-client/src/lib.rs (hunk 1 above)
Comment (mrwwheat7) [reply etcd-client/src/lib.rs]: Done

File: etcd-client/src/lib.rs (hunk 2 above)
Comment (mrwwheat7) [reply etcd-client/src/lib.rs]: Done

**Review by pwang7 — APPROVED** [review]

**Pull request merged**